python/mspasspy/db/database.py

```diff
@@ -249,33 +343,26 @@ def _sync_metadata_before_update(self, d):
             d.put_string('time_standard', 'relative')
         else:
             d.put_string('time_standard', 'UTC')
-        if 'dtype' not in d:
-            if isinstance(d, TimeSeries):
-                d['dtype'] = 'TimeSeries'
-            elif isinstance(d, Seismogram):
-                d['dtype'] = 'Seismogram'
-            else:
-                raise TypeError("only TimeSeries and Seismogram are supported")
 
     def _save_history(self, mspass_object, collection='history_object'):
         history_col = self[collection]
         history_binary = pickle.dumps(ProcessingHistory(mspass_object))
         # todo jobname jobid
-        if 'history_id' in mspass_object:
+        if 'history_object_id' in mspass_object:
```

Comment:
Instead of hard-code in the name, we should probably replace `'history_object_id'` with `collection+'_id'`. There are quite a few similar issues throughout this module.